File: ToobarTest/app/src/main/java/com/example/toolbartest/MainActivity.java
package com.example.toolbartest;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

public class MainActivity extends AppCompatActivity {
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView();
    }

    private void initView() {
        toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        toolbar.setTitle(R.string.homepage);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.toolbar_action, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch(item.getItemId()){
            case R.id.toolbar_search:
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: SwordDemoLibrary/camera2/src/main/java/com/example/camera2/MainActivity.kt
package com.example.camera2

import android.content.Context
import android.graphics.ImageFormat
import android.hardware.camera2.CameraCaptureSession
import android.hardware.camera2.CameraDevice
import android.media.ImageReader
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.DisplayMetrics
import android.util.TypedValue
import android.view.Surface
import android.view.SurfaceView
import android.view.WindowManager

class MainActivity : AppCompatActivity() {
    private lateinit var cameraDevice: CameraDevice
    private lateinit var surfaceView: SurfaceView
    private lateinit var imageReader: ImageReader
    private lateinit var displayMetrics: DisplayMetrics

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        obtainMetrics()
        
        surfaceView = findViewById(R.id.preview_view)
//        imageReader = ImageReader.newInstance(50.dp, 50.dp, ImageFormat.JPEG, 1)
    }

    

    private fun createCaptureSession(surfaces: List<Surface>) {
        cameraDevice.createCaptureSession(surfaces, object : CameraCaptureSession.StateCallback() {
            override fun onConfigured(session: CameraCaptureSession) {
                TODO("Not yet implemented")
            }

            override fun onConfigureFailed(session: CameraCaptureSession) {
                TODO("Not yet implemented")
            }

        }, null)
    }

    fun Int.dp(): Int =
        TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, this.toFloat(), displayMetrics).toInt()

    private fun obtainMetrics() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            display?.getMetrics(displayMetrics)
        } else {
            (getSystemService(Context.WINDOW_SERVICE) as WindowManager).defaultDisplay.getMetrics(displayMetrics)
        }
    }
}

